packages/jest-matcher-utils/src/index.ts

```diff
@@ -130,42 +130,51 @@ export const ensureNoExpected = (
   }
 };
 
+/**
+ * Ensures that `actual` is of type `number | bigint`
+ */
 export const ensureActualIsNumber = (
   actual: unknown,
   matcherName: string,
   options?: MatcherHintOptions,
 ) => {
-  if (typeof actual !== 'number') {
+  if (typeof actual !== 'number' && typeof actual !== 'bigint') {
     // Prepend maybe not only for backward compatibility.
     const matcherString = (options ? '' : '[.not]') + matcherName;
     throw new Error(
       matcherErrorMessage(
         matcherHint(matcherString, undefined, undefined, options),
-        `${RECEIVED_COLOR('received')} value must be a number`,
+        `${RECEIVED_COLOR('received')} value must be a number or bigint`,
         printWithType('Received', actual, printReceived),
       ),
     );
   }
 };
 
+/**
+ * Ensures that `expected` is of type `number | bigint`
+ */
 export const ensureExpectedIsNumber = (
   expected: unknown,
   matcherName: string,
   options?: MatcherHintOptions,
 ) => {
-  if (typeof expected !== 'number') {
+  if (typeof expected !== 'number' && typeof expected !== 'bigint') {
     // Prepend maybe not only for backward compatibility.
     const matcherString = (options ? '' : '[.not]') + matcherName;
     throw new Error(
       matcherErrorMessage(
         matcherHint(matcherString, undefined, undefined, options),
-        `${EXPECTED_COLOR('expected')} value must be a number`,
+        `${EXPECTED_COLOR('expected')} value must be a number or bigint`,
         printWithType('Expected', expected, printExpected),
       ),
     );
   }
 };
 
+/**
+ * Ensures that `actual` & `expected` are of type `number | bigint`
```

Comment:
I think we need to leave the original 3 exported functions as they were and provide 3 new functions, as you had done originally, because it seems incorrect for some existing matchers /cc @SimenB @jeysal what do you think?

Evidence is updated snapshots for matchers errors in `spyMatchers.test.ts.snap`

Reply:
The reason of having 3 separate  checkers was due to `toBeCloseTo` originally not having bigint support, since it does now, where will the additional checkers be used if the numeric matchers allow mixed types? Since it seems `createToBeCalledTimesMatcher` is the only other matcher that uses 1 of 3 number checker, Should we create a single checker for it? Or better yet, does it make more since to use `ensureExpectedIsNonNegativeInteger` for `createToBeCalledTimesMatcher`

Reply:
In theory, packages outside Jest monorepo could depend on current behavior.

Having said that, no matchers in `jest-community/jest-extended` package call them.

So I withdraw this suggestion to keep moving forward (see summary comment in issue)

And yes, as soon as breaking-change phase of Jest 25 starts, PRs are welcome:

* define `ensureExpectedIsNonNegativeInteger` and call it in 2 spy matchers
* throw matcher error if `precision` is not an integer in `toBeCloseTo` (because the message needs to be specific about `precision` instead of `expected` we cannot write a generic ensure function for it)

Reply:
> Or better yet, does it make more since to use `ensureExpectedIsNonNegativeInteger` for `createToBeCalledTimesMatcher`

Sorry, suggested `ensureExpectedIsNonNegativeInteger` because its already defined and used in `toHaveLength`

Reply:
Super! So we can replace `ensureExpectedIsNumber` with `ensureExpectedIsNonNegativeInteger` in spy matchers.